pkg/cmd/cli/secrets/add_secret_to_obj.go

```diff
@@ -61,14 +59,14 @@ type AddSecretOptions struct {
 // NewCmdAddSecret creates a command object for adding a secret reference to a service account
 func NewCmdAddSecret(name, fullName string, f *cmdutil.Factory, out io.Writer) *cobra.Command {
 	o := &AddSecretOptions{Out: out}
-	typeFlag := "mount"
+	typeFlags := util.StringList{"mount"}
```

Comment:
This is the `Set` function for `StringList`:
```
func (sl *StringList) Set(value string) error {
	for _, s := range strings.Split(value, ",") {
		*sl = append(*sl, s)
	}
	return nil
}
```
It is only additive. that's why `--for=mount --for=pull` behaves the same as `--for=mount,pull`. It also means you can't set a default value and have it overridden by a flag